Redefinitions causing code to fail compilation on Sparkfun Artemis Boards (Apollo 3)

## Bug Report

### Arduino Platform
* [x] Apollo3 board


<!-- Please specify board name if not included in platform selection -->

### IDE
* [x] Arduino IDE


### IR-Protocol
* [x] Other - Not relavent

### Example to reproduce the issue
* [x] AllProtocols
* [x] SendAllProtocols 
* [ ] Callback
* [ ] Interrupt
* [ ] OneProtocol
* [ ] ReceiveAndSend
* [ ] ReceiveAndSendDynamicPins
* [ ] RFProtocols
* [ ] SimpleReceiver
* [ ] SimpleSender
* [ ] Other

### Pin(s) used for IR-receive, if not default

### Version
* [x] Yes I use the [latest release](https://github.com/ukw100/IRMP/releases) and verified this!
* [ ] Other - please specify

### Current behavior
<!-- Paste the code and output you run -->
Code is the AllProtocols example with no changes made to it

<details>
<summary>Output</summary>

```
In file included from D:\Documents\Arduino\libraries\IRMP\src/irmp.h:31,
                 from D:\Documents\Arduino\libraries\IRMP\src/irmp.c.h:26,
                 from C:\Users\Gabe\AppData\Local\Temp\arduino_modified_sketch_266082\AllProtocols.ino:88:
D:\Documents\Arduino\libraries\IRMP\src/irmpArduinoExt.h:84: warning: "F" redefined
 #define F(a) a
 
In file included from C:\Users\Gabe\AppData\Local\Arduino15\packages\SparkFun\hardware\apollo3\2.0.3/cores/arduino/mbed-bridge/core-api/api/Print.h:24,
                 from C:\Users\Gabe\AppData\Local\Arduino15\packages\SparkFun\hardware\apollo3\2.0.3/cores/arduino/mbed-bridge/core-api/api/Stream.h:25,
                 from C:\Users\Gabe\AppData\Local\Arduino15\packages\SparkFun\hardware\apollo3\2.0.3/cores/arduino/mbed-bridge/core-api/api/Client.h:22,
                 from C:\Users\Gabe\AppData\Local\Arduino15\packages\SparkFun\hardware\apollo3\2.0.3/cores/arduino/mbed-bridge/core-api/api/ArduinoAPI.h:29,
                 from C:\Users\Gabe\AppData\Local\Arduino15\packages\SparkFun\hardware\apollo3\2.0.3/cores/arduino/mbed-bridge/Arduino.h:14,
                 from C:\Users\Gabe\AppData\Local\Arduino15\packages\SparkFun\hardware\apollo3\2.0.3/cores/arduino/sdk/ArduinoSDK.h:9,
                 from <command-line>:
C:\Users\Gabe\AppData\Local\Arduino15\packages\SparkFun\hardware\apollo3\2.0.3/cores/arduino/mbed-bridge/core-api/api/String.h:45: note: this is the location of the previous definition
 #define F(string_literal) (reinterpret_cast<const __FlashStringHelper *>(PSTR(string_literal)))
 
c:/users/gabe/appdata/local/arduino15/packages/sparkfun/tools/arm-none-eabi-gcc/8-2018-q4-major/bin/../lib/gcc/arm-none-eabi/8.2.1/../../../../arm-none-eabi/bin/ld.exe: C:\Users\Gabe\AppData\Local\Arduino15\packages\SparkFun\hardware\apollo3\2.0.3\variants\SFE_ARTEMIS_ATP/mbed/libmbed-os.a(isr.o): in function `am_ctimer_isr':
/github/workspace/mbed-builds/artemis-redboard-atp-lib_1/mbed-os/targets/TARGET_Ambiq_Micro/TARGET_Apollo3/device/isr.c:6: multiple definition of `am_ctimer_isr'; sketch\AllProtocols.ino.cpp.o:D:\Documents\Arduino\libraries\IRMP\src/IRTimer.cpp.h:657: first defined here
collect2.exe: error: ld returned 1 exit status
exit status 1
Error compiling for board RedBoard Artemis ATP.

```
</details>
<!-- Add a clear and concise description of the behavior. -->
Code fails during compilation due to redefinition of `am_ctimer_isr(void)` and `F(a)`

### Expected behavior
<!-- Add a clear and concise description of what you expected to happen. -->
Code Compiles correctly. 

### Additional context

<!-- (Optional) Add any other context about the problem here. -->
It seems that Sparkfun added their own functions that implement the interrupts and `F(a)` substitution. 
* **Interrupt** function:  Identical to the interrupt function that that this repository has implemented.
* **Flash String Helper**: Differs from this repo's implementation by using the true Arduino implementation. 

I tested commenting out the function definitions of `am_ctimer_isr(void)` and `F(a)` in this library and found that the Arduino IDE properly compiles the `AllProtocols` example when the configured board is the Sparkfun Artemis Redboard ATP. I will plan to verify that the library works properly with the code commented out, and will submit a PR if I find that it does.

--- a/src/IRTimer.cpp.h
+++ b/src/IRTimer.cpp.h
@@ -653,20 +653,6 @@ void enableIRTimerInterrupt(void)
  * We use TIMER2_COMPB_vect to be compatible with tone() library
  */
 
-//#if defined(ARDUINO_ARCH_APOLLO3)
-///*
-// * The isr, which calls the dispatcher/service, which call the irmp_timer_ISR
-// */
-//extern "C" void am_ctimer_isr(void) {
-//    // Check and clear any active CTIMER interrupts.
-//    uint32_t ui32Status = am_hal_ctimer_int_status_get(true);
-//    am_hal_ctimer_int_clear(ui32Status);
-//
-//    // Run handlers for the various possible timer events.
-//    am_hal_ctimer_int_service(ui32Status);
-//}
-//#endif
-
 #if defined(__AVR__)
 
 #  if F_CPU < 8000000L

--- a/src/irmpArduinoExt.h
+++ b/src/irmpArduinoExt.h
@@ -80,10 +80,6 @@
 
 #elif defined(ARDUINO_ARCH_MBED) // Arduino Nano 33 BLE + Sparkfun Apollo3
 #include "mbed.h"
-// F is undefined in mbed.h, so F() is unknown and leads to "'F' was not declared in this scope" errors. -> define it again.
-#ifndef F
-#define F(a) a
-#endif
 
 #elif defined(ESP32)
 #elif defined(STM32F1xx) // for "Generic STM32F1 series" from "STM32 Boards (selected from submenu)" of Arduino Board manager

--- a/src/irsndArduinoExt.h
+++ b/src/irsndArduinoExt.h
@@ -60,10 +60,6 @@ void irsnd_data_print(Print *aSerial, IRMP_DATA *aIRMPDataPtr);
 
 #if defined(ARDUINO_ARCH_MBED) // Arduino Nano 33 BLE + Sparkfun Apollo3
 #include "mbed.h"
-// F is undefined in mbed.h, so F() is unknown and leads to "'F' was not declared in this scope" errors. -> define it again.
-#ifndef F
-#define F(a) a
-#endif
 #define F_CPU 0 // dummy definition to avoid warning at irsnd.c.h:27 #error F_CPU unkown
 #endif
 

